### DISCUSSION TITLE
Delete all photos for a specific user

### DISCUSSION THREAD

**Discussion by alez007** [immich-app/immich#1562] [Q&A]
Hi, is there a way to remove all photos of a specific user ? Empty the library effectively. 

I've imported allot of pictures but using the wrong user and I don't want to select day by day then delete, there seems to be missing a select all feature or I'm simply not seeing it, either in the web app or mobile app.

Thanks

**Comment by alextran1502** [comment]
Hello, right now we don't have this feature yet. The only delete all is delete the user and all the associated data.

**Comment by MichaelBui** [comment]
You can do it via APIs if you have access to DB (a bit technical):
- Get the user ID from `users` table
- Get the API key for that user (better to do it in web UI)
- Get all the asset ID from `assets` table
- Call the `DELETE /asset` API with given IDs (in the proper structure)
Remember to always do a backup before this if you're not confident about doing this kind of stuff

**Reply by alez007** [reply]
hehe if I'm gonna do a script to delete everything I might as well submit a PR with a delete assets functionality... just wondering if there's a reason why this functionality is not there yet, seems something obvious to me

**Reply by bo0tzz** [reply]
The reason it's not there is because Immich is still under development ;) A PR would be welcome! The most straightforward way is probably just to add a "select all" button to the UI.

**Reply by sforli** [reply]
I like the "Select all" button.

In the meantime, what would be the sequence of commands to access the DB thought the API and wipe everything?

**Reply by bo0tzz** [reply]
Note, select all is now supported in the web UI.

**Reply by cjrodriguez98** [reply]
Did not found the select all in the web UI....

**Reply by rochiey** [reply]
@bo0tzz where?

**Reply by bo0tzz** [reply]
Ctrl+A, or select something and then use this button
![image](https://github.com/user-attachments/assets/e4bef36c-a10d-4b94-913f-5bcace5eef5f)

**Reply by rochiey** [reply]
If you hadn't said ctrl + a, i would have never guessed there's a select all feature. No button for select all is diabolical.
